src/state/pportfolio/hooks.ts

```diff
@@ -0,0 +1,231 @@
+        // If token2 does not exist its because this pair is not a pair but a single staking
+        if (!token2) {
+          return new TokenDataUser(
+            new Token(
+              chainId,
+              ethers.utils.getAddress(token1?.id),
+              token1?.decimals,
+              `${token1?.symbol} - Staked`,
+              token1?.name,
+            ),
+            token1?.price,
+            token1?.amount,
+          );
+        }
+
+        const tokenA = new TokenAmount(
+          new Token(chainId, ethers.utils.getAddress(token1?.id), token1?.decimals, token1?.symbol, token1?.name),
+          token1?.amount.toString().replace('.', ''),
+        );
+
+        const tokenB = new TokenAmount(
+          new Token(chainId, ethers.utils.getAddress(token2?.id), token2?.decimals, token2?.symbol, token2?.name),
+          token2?.amount.toString().replace('.', ''),
+        );
+
+        return new PairDataUser(new Pair(tokenA, tokenB, chainId), pair?.stats?.net_usd_value, undefined, [
+          token1?.logo_url,
+          token2?.logo_url,
+        ]);
+      });
+      return requestPairs;
+    }
+    return [];
+  };
+
+  const getBalance = async () => {
+    if (account) {
+      if (!chain || !chain.mainnet) {
+        chain = AVALANCHE_MAINNET;
+      }
+
+      const query = qs.stringify(
+        {
+          id: account,
+          chain_id: chain.symbol.toLowerCase(),
+        },
+        {
+          encodeValuesOnly: true,
+        },
+      );
+
+      const response = await openApi.get(`/token_list?${query}`);
+      const data = response.data;
+
+      const requestTokens: TokenDataUser[] = data.map((token: any) => {
+        if (token?.id?.toLowerCase() === 'avax') {
+          return new TokenDataUser(CAVAX[chainId], token?.price, token?.amount);
+        }
+
+        if (!isAddress(token?.id)) {
+          return new TokenDataUser(
+            new Token(chainId, ZERO_ADDRESS, token?.decimals, token?.symbol, token?.name),
+            token?.price,
+            token?.amount,
+            undefined,
+            token?.logo_url, // work around now for other coins of other chains
+          );
+        }
+
+        return new TokenDataUser(
+          new Token(chainId, ethers.utils.getAddress(token?.id), token?.decimals, token?.symbol, token?.name),
+          token?.price,
+          token?.amount,
+          undefined,
+          token?.logo_url,
+        );
+      });
+
+      if (chainId === ChainId.AVALANCHE) {
+        const pairs = await getPangolinPairs();
+        return [...requestTokens, ...pairs];
+      }
+
+      return requestTokens;
+    }
+    return null;
+  };
+
+  return useQuery(
+    ['getWalletChainTokens', chainId.toString(), account],
+    async () => {
+      const tokens = await getBalance();
+
+      if (!tokens) {
+        return undefined;
+      }
+
+      tokens.sort((a, b) => b.usdValue - a.usdValue);
+
+      return tokens.filter((token) => token.usdValue >= 0.01);
```

Comment:
lets do sorting and filtering in `getBalance` so useQuery just returns whatever given by getBalance